### PR TITLE
core: frontend: Add menu items to home

### PR DESCRIPTION
**Pull request by Williangalvani**
home

### CODE REVIEW

File: core/frontend/src/views/MainView.vue [hunk 1]
```diff
@@ -18,13 +18,108 @@
...
+      const items: menuItem[] = []
+      for (const item of this.menus) {
+        if (item.submenus !== undefined) {
+          for (const subitem of item.submenus) {
+            if (!subitem.advanced || this.settings.is_pirate_mode) {
+              items.push(subitem)
+            }
+          }
+        }
+      }
```
Comment (patrickelectric) [inline core/frontend/src/views/MainView.vue:104]: Can you just do:
```suggestion
      for (const item of this.menus) {
        for (const subitem of item.submenus) {
          if (!subitem?.advanced || this.settings.is_pirate_mode) {
            items.push(subitem)
          }
        }
      }
```

File: core/frontend/src/menus.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        text: 'Allow browsing the Telemetry (.tlog) and Binary (.bin) logs generated by your vehicle. Bin logs are'
+              + ' currently only supported for Navigator boards',
+      },
+      {
+        title: 'Endpoints',
+        icon: 'mdi-arrow-decision',
+        route: '/autopilot/endpoints',
+        advanced: true,
+        text: 'Manage Mavlink endpoints for internal/external systems. Used this if you need to connect additional'
+              + ' mavlink systems to your vehicle',
```
Comment (ES-Alexander) [inline core/frontend/src/menus.ts:40]:
```suggestion
        text: 'Manage MAVLink endpoints for internal/external systems. Use this if you need to connect additional'
              + ' MAVLink systems to your vehicle',
```

File: core/frontend/src/menus.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        advanced: true,
+        text: 'Browse all the files in BlueOS, useful for fetching logs,'
+              + ' tweaking configurations, and development',
+      },
+      {
+        title: 'NMEA Injector',
+        icon: 'mdi-map-marker',
+        route: '/tools/nmea-injector',
+        advanced: true,
+        text: 'Used for forwarding UDP NMEA stream into Ardupilot',
```
Comment (ES-Alexander) [inline core/frontend/src/menus.ts:84]:
```suggestion
        text: 'Used for forwarding UDP NMEA streams into Ardupilot',
```
Or should this be one? ("a UDP NMEA stream")

File: core/frontend/src/menus.ts [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        route: '/tools/system-information',
+        advanced: false,
+        text: 'Detailed system status information, CPU, memory, disk, and ethernet status',
+      },
+      {
+        title: 'Network test',
+        icon: 'mdi-speedometer',
+        route: '/tools/network-test',
+        show: true,
+        text: 'Test link speed between topside computer and your vehicle',
```
Comment (ES-Alexander) [inline core/frontend/src/menus.ts:98]: I like this for sub, although I'm assuming drone users don't refer to the GCS as a "topside" computer? Not sure what our preference is here (do we want default sub terminology, or actively keep things general to ArduPilot vehicles?)

File: core/frontend/src/menus.ts [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      },
+      {
+        title: 'NMEA Injector',
+        icon: 'mdi-map-marker',
+        route: '/tools/nmea-injector',
+        advanced: true,
+        text: 'Used for forwarding UDP NMEA stream into Ardupilot',
+      },
+      {
+        title: 'System information',
```
Comment (ES-Alexander) [inline core/frontend/src/menus.ts:87]:
```suggestion
        title: 'System Information',
```

File: core/frontend/src/menus.ts [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+        icon: 'mdi-chart-pie',
+        route: '/tools/system-information',
+        advanced: false,
+        text: 'Detailed system status information, CPU, memory, disk, and ethernet status',
+      },
+      {
+        title: 'Network test',
```
Comment (ES-Alexander) [inline core/frontend/src/menus.ts:94]:
```suggestion
        title: 'Network Test',
```

File: core/frontend/src/menus.ts [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+      },
+      {
+        title: 'MAVLink Inspector',
+        icon: 'mdi-chart-areaspline',
+        route: '/tools/mavlink-inspector',
+        advanced: true,
+        text: 'View detailed MAVLink traffic coming from your vehicle',
+      },
+      {
+        title: 'Version-chooser',
```
Comment (ES-Alexander) [inline core/frontend/src/menus.ts:115]:
```suggestion
        title: 'Version Chooser',
```

File: core/frontend/src/menus.ts [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+        icon: 'mdi-cellphone-arrow-down',
+        route: '/tools/version-chooser',
+        advanced: false,
+        text: 'Manage versions and update to latest available',
```
Comment (ES-Alexander) [inline core/frontend/src/menus.ts:119]:
```suggestion
        text: 'Manage BlueOS versions and update to the latest available',
```

**Review by ES-Alexander — CHANGES_REQUESTED** [review]

File: core/frontend/src/menus.ts [hunk 9]
```diff
@@ -0,0 +1,134 @@
...
+        text: 'General Autopilot settings, allow you to start/stop Ardupilot if using Navigator or SITL, and switch'
+              + ' boards if more than one is present',
+      },
+      {
+        title: 'Firmware',
+        icon: 'mdi-chip',
+        route: '/autopilot/firmware',
+        advanced: false,
+        text: 'Used to download and flash new firmware to your board',
+      },
```
[the remark this answers is not in the document]
Comment (Williangalvani) [reply core/frontend/src/menus.ts:25]: the change is trivial to do after we merge them, I think this isn't an issue. I renamed to "current flight controller"

File: core/frontend/src/menus.ts (hunk 4 above)
Comment (Williangalvani) [reply core/frontend/src/menus.ts:98]: we're currently walking the line... I don't want to be in any sides too much :sweat_smile: 
see my previous comment about "flight controller"

File: core/frontend/src/views/MainView.vue (hunk 1 above)
Comment (Williangalvani) [reply core/frontend/src/views/MainView.vue:102]: not all of them have submenus defined, typescript complains about it without the check. I did something else you may like, tho...

File: core/frontend/src/menus.ts (hunk 4 above)
Comment (Williangalvani) [reply core/frontend/src/menus.ts:97]: I can just remove the "topside" reference

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**